configure.ac

```diff
@@ -144,7 +144,7 @@ AC_DEFINE_UNQUOTED([OFI_NCCL_TRACE], [${trace}], [Defined to 1 unit test output
 AC_ARG_ENABLE([picky-compiler],
    [AS_HELP_STRING([--disable-picky-compiler], [Disable adding picky compiler flags.])])
 AS_IF([test "${enable_picky_compiler}" != "no"],
-      [picky_compiler_flags="-Wall -Wc++-compat -Wextra -Wno-unused-parameter"
```

Comment:
Any harm leaving this?

Reply:
GCC warns when this flag is passed on C++ source files, as this warning is C specific.